File: WallBreakers/SimpleBitManipulation/HammingDistance.java
class Solution {
  public int hammingDistance(int x, int y) {
    int dist = 0;
    x ^= y;
    while (x>0) {
      x &= (x-1); // set rightmost 1 to 0
      dist++;
    }
    return dist;
  }
  // public int hammingDistance(int x, int y) {
  //   return Integer.bitCount(x^y);
  // }
}

File: WallBreakers/SimpleBitManipulation/HammingDistance.py
class Solution:
  def hammingDistance(self, x: int, y: int) -> int:
    diff, count = x^y, 0
    while diff > 0:
      diff &= (diff-1) # set rightmost bit 1 to 0
      count += 1
    return count
    # return bin(x^y).count("1");